### PR TITLE
feat(BCP20-850) add in customer impersonation token

### PR DESCRIPTION
**Pull request by jivewise**
Add in customer impersonation token retrieval for on login.

### CODE REVIEW

File: src/api/customers/handlers/get-logged-in-customer.ts [hunk 1]
```diff
@@ -21,24 +23,31 @@ export const getLoggedInCustomerQuery = /* GraphQL */ `
...
   res,
   config,
 }) => {
-  const token = req.cookies[config.customerCookie]
+  // use the session if available, otherwise fall back on the customerCookie
+  const session: IronSession = await getIronSession(req, res, sessionOptions)
+  const token: string = session.token || req.cookies[config.customerCookie]
+  const headers: HeadersInit = session.token
+    ? {
+        Authorization: `Bearer ${session.token}`,
```
Comment (zhangpengchen) [inline src/api/customers/handlers/get-logged-in-customer.ts:38]: Is BigCommerce API already supported customer impersonation jwt token?

_Request not allowed: Detected that request originated from a browser script and is using a customer impersonation jwt token. Customer impersonation tokens are not allowed with frontend requests._

Comment (zhangpengchen) [reply src/api/customers/handlers/get-logged-in-customer.ts:38]: Sorry to ask here as I am working on the same problem and saw this valuable pull request, couldn't figure out where to pass the customer impersonation jwt token I received : )

File: src/api/operations/get-customer-impersonation-token.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+async function getCustomerImpersonationToken({
+  config,
+}: {
+  config?: BigcommerceConfig
+} = {}): Promise<string> {
+  config = getConfig(config)
+  const options = {
+    method: 'POST',
+    body: JSON.stringify({
+      channel_id: config.storeChannelId || 1,
```
Comment (Vinsanity) [inline src/api/operations/get-customer-impersonation-token.ts:13]: They still require a channel ID for this?

**Review by Vinsanity — APPROVED** [review]
Looks good here Jing, just a question about channel id, nut no blockers.

File: src/api/customers/handlers/get-logged-in-customer.ts (hunk 1 above)
Comment (zhangpengchen) [reply src/api/customers/handlers/get-logged-in-customer.ts:38]: Never mind, including a SHOP_TOKEN cookie in Postman results in 403 Forbidden `Request not allowed`.

Comment (zhangpengchen) [reply src/api/customers/handlers/get-logged-in-customer.ts:38]: Yes, it works

**Review by jorgemasta — APPROVED** [review]

**Pull request merged**